amlb/results.py

```diff
@@ -671,67 +671,104 @@ class TimeSeriesResult(RegressionResult):
 
     def __init__(self, predictions_df, info=None):
         super().__init__(predictions_df, info)
-        self.truth = self.df['truth'].values if self.df is not None else None #.iloc[:, 1].values if self.df is not None else None
-        self.predictions = self.df['predictions'].values if self.df is not None else None #.iloc[:, -2].values if self.df is not None else None
-        self.y_past_period_error = self.df['y_past_period_error'].values
-        self.quantiles = self.df.iloc[:, 2:-1].values
-        self.quantiles_probs = np.array([float(q) for q in self.df.columns[2:-1]])
-        self.truth = self.truth.astype(float, copy=False)
-        self.predictions = self.predictions.astype(float, copy=False)
-        self.quantiles = self.quantiles.astype(float, copy=False)
-        self.y_past_period_error = self.y_past_period_error.astype(float, copy=False)
+        self.quantiles_probs_str = list(column for column in self.df.columns if column.startswith('0.'))
+        if not all(x in self.df.columns for x in ['truth', 'predictions', 'item_id', 'seasonal_error']) or len(self.quantiles_probs_str) == 0:
+            msg=f'Missing columns for calculating time series metrics. Given columns are {list(self.df.columns)}.'
+            raise ValueError(msg)
+
+        self.truth = self.df['truth'].values if self.df is not None else None
+        self.pred_mean = self.df['predictions'].values if self.df is not None else None
+
+        self.seasonal_error = self.df['seasonal_error'].values
+        self.quantiles_probs = np.array(list(float(quantile_prob_str) for quantile_prob_str in self.quantiles_probs_str))
+        self.quantiles = self.df.filter(self.quantiles_probs_str).values
+        self.item_ids = self.df['item_id'].values
+        _, unique_item_ids_counts = np.unique(self.item_ids, return_counts=True)
+        if np.sum(np.abs(np.mean(unique_item_ids_counts) - unique_item_ids_counts)) != 0.:
+            msg = f'Error: Predicted sequences have different lengths {unique_item_ids_counts}.'
+            raise ValueError(msg)
+        self.prediction_length = unique_item_ids_counts[0]
+        self.unique_item_ids = self.item_ids.copy()[::self.prediction_length]
+
+        self.dtype = float
+        self.truth = self.truth.astype(self.dtype, copy=False)
+        self.pred_mean = self.pred_mean.astype(self.dtype, copy=False)
+        self.quantiles = self.quantiles.astype(self.dtype, copy=False)
```

Comment:
Why are these separate from the assignments above? Is this more performant than keeping them in a dataframe? To me it seems like they are mostly just used as aliases. Doesn't the dataframe typing generally provide more readable code? E.g.,
```python
     def itemwise_mean(self, values):
         return np.array([np.mean(values[self.item_ids == unique_item_id]) for unique_item_id in self.unique_item_ids], dtype=self.dtype)

     def itemwise_select_first(self, values):
         return values.copy()[::self.prediction_length]
```
to
```python
     def itemwise_mean(self, column):
         return self.df.groupby("item_ids")[column].mean()

     def itemwise_select_first(self, column):
         return self.df.groupby("item_ids")[column].first()
```